### PR TITLE
Add named ucr expression

### PR DESCRIPTION
**Pull request by czue**
http://manage.dimagi.com/default.asp?184085

This also changes the spec to allow any string/integer/float/boolean constant in an expression to resolve to a constant expression, which should hopefully make specs a bit easier to read/write

### CODE REVIEW

File: corehq/apps/userreports/expressions/specs.py [hunk 1]
```diff
@@ -23,7 +23,13 @@ def __call__(self, item, context=None):
...
 class ConstantGetterSpec(JsonObject):
     type = TypeProperty('constant')
-    constant = DefaultProperty(required=True)
+    constant = DefaultProperty()
+
+    @classmethod
+    def wrap(self, obj):
+        if 'constant' not in obj:
+            raise BadSpecError('"constant" property is required!')
+        return super(ConstantGetterSpec ,self).wrap(obj)
```
Comment (dimagimon) [inline corehq/apps/userreports/expressions/specs.py]: E203 whitespace before ','

Comment (dimagimon) [inline corehq/apps/userreports/expressions/specs.py]: E231 missing whitespace after ','

**Comment by czue** [comment]
@NoahCarnahan

**Comment by czue** [comment]
:white_check_mark:

**Comment by NoahCarnahan** [comment]
Looks good. I do think that the name "named expression" is a bit confusing though. It sounds like maybe this would work in a similar way to "named filters", but that's not the case. I won't block on that though since this is time sensitive.

**Comment by NoahCarnahan** [comment]
Also, love the constant literals!

**Comment by czue** [comment]
good point - will rename/tweak tomorrow

**Pull request merged**